src/components/popup-menu/popup-menu.jsx

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import injectSheet from 'react-jss';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import styles from './popup-menu.styles';
+import { Icon } from '../../';
+
+class PopupMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: props.isOpen,
+    };
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClick);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      this.setState({
+        isOpen: nextProps.isOpen,
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick);
+  }
+
+  handleClick = ({ target } = {}) => {
+    if (target && this.onOutsideElement && !this.onOutsideElement.contains(target)) {
+      this.handleClickOutside();
+    }
+  };
+
+  handleClickOutside = () => {
+    if (this.state.isOpen) {
+      this.toggleMenu(false);
+    }
+  };
+
+  toggleMenu = (forceOpenToggle = null) => {
+    const isOpen = forceOpenToggle !== null ? forceOpenToggle : !this.state.isOpen;
+    this.setState({
+      isOpen,
+    });
+    this.props.onToggle(isOpen);
+  };
+
+  render() {
+    const { width, toggleButton, classes, children, enter, exit } = this.props;
+    const { isOpen } = this.state;
+    return (
+      <span
+        className={classes.menuContainer}
+        ref={element => {
+          this.onOutsideElement = element;
+        }}
+      >
+        <button className={classes.menuButton} onClick={this.toggleMenu}>
+          {toggleButton || <Icon.VerticalEllipsis height={28} width={28} stroke="currentColor" fill="currentColor" />}
+        </button>
+        <TransitionGroup>
+          {isOpen && (
+            <CSSTransition classNames={classes.menuSlideDown} timeout={{ exit, enter }}>
+              <div className={classes.menuPopup} style={{ width }}>
+                <div className={classes.menuItemContainer}>
+                  <div>
+                    <ul className={classes.menuItems}>{children}</ul>
+                  </div>
+                </div>
+              </div>
+            </CSSTransition>
+          )}
+        </TransitionGroup>
+      </span>
+    );
+  }
+}
+
+PopupMenu.propTypes = {
+  width: PropTypes.number,
```

Comment:
Would it make sense that this could be a string as well? (I am thinking about these cases: `auto` or `100%`)

Reply:
Absolutely correct.